amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/fanout/FanOutRecordsPublisher.java

```diff
@@ -681,6 +698,7 @@ public BatchUniqueIdentifier batchUniqueIdentifier() {
         private boolean isDisposed = false;
         private boolean isErrorDispatched = false;
         private boolean isCancelled = false;
+        private RequestDetails recordFlowLastSuccessfuRequestDetails;
```

Comment:
why this needs to be a class variable? Let's keep the scope of this object to be within the responceReceived() method.

Reply:
Yeah this doesn't need to be class level, will move it to be scoped within `responseReceived()`.